sdk/tables/data-tables/src/TableBatchPolicies.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  BaseRequestPolicy,
+  RequestPolicy,
+  RequestPolicyOptions,
+  WebResourceLike,
+  HttpOperationResponse,
+  RequestPolicyFactory,
+  WebResource,
+  HttpHeaders
+} from "@azure/core-http";
+import { HeaderConstants } from "./utils/constants";
+import { InnerBatchRequest } from "./utils/internalModels";
+
+class BatchHeaderFilterPolicy extends BaseRequestPolicy {
+  // eslint-disable-next-line @typescript-eslint/no-useless-constructor
+  constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+    super(nextPolicy, options);
+  }
+  public async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
+    request.headers.remove(HeaderConstants.X_MS_VERSION); // The subrequests should not have the x-ms-version header.
+    return this._nextPolicy.sendRequest(request);
+  }
+}
+
+export class BatchHeaderFilterPolicyFactory implements RequestPolicyFactory {
+  // eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
+  public create(nextPolicy: RequestPolicy, options: RequestPolicyOptions): BatchHeaderFilterPolicy {
```

Comment:
Personally I prefer to have factories return the base type of `RequestPolicyFactory` because then we don't have to expose the policy constructor directly if we ever want to expose the factory.

It's not a big deal since you don't actually export either right now, but might be helpful if these ever get moved elsewhere.